test/integration/node-specific/resource_clean_up.test.ts

```diff
@@ -0,0 +1,90 @@
+import { expect } from 'chai';
+
+import { runScript } from './resource_tracking_script_builder';
+
+/**
+ * This 5MB range is selected arbitrarily and should likely be raised if failures are seen intermittently.
+ *
+ * The goal here is to catch unbounded memory growth, currently runScript defaults to 100 iterations of whatever test code is passed in
+ * More than a 5MB growth in memory usage after the script has finished and _should have_ cleaned up all its resources likely indicates that
+ * the growth will continue if the script is changed to iterate the the test code more.
+ */
+const MB_PERMITTED_OFFSET = 5;
+
+describe('Driver Resources', () => {
+  let startingMemoryUsed;
+  let endingMemoryUsed;
+  let heap;
+
+  beforeEach(function () {
+    if (globalThis.AbortController == null) {
+      if (this.currentTest) this.currentTest.skipReason = 'Need AbortController to run this test';
+      this.currentTest?.skip();
+    }
+    if (typeof this.configuration.serverApi === 'string') {
+      if (this.currentTest) {
+        this.currentTest.skipReason = 'runScript does not support serverApi settings';
+      }
+      this.currentTest?.skip();
+    }
+  });
+
+  before('create leak reproduction script', async function () {
+    if (globalThis.AbortController == null || typeof this.configuration.serverApi === 'string') {
+      return;
+    }
+    const res = await runScript(
+      'no_resource_leak_connect_close',
+      this.configuration,
+      async function run({ MongoClient, uri }) {
+        const mongoClient = new MongoClient(uri);
+        await mongoClient.connect();
+        const db = mongoClient.db();
+        await Promise.all([
+          db.collections(),
+          db.collections(),
+          db.collections(),
+          db.collections(),
+          db.collections(),
+          db.collections(),
+          db.collections(),
+          db.collections()
+        ]);
+        await mongoClient.close();
+      }
+    ).catch(error => error);
+
+    if (res instanceof Error) {
```

Comment:
Sounds good, I can do that, the expectation is that the script never exits with failure so this is just to help the engineer get pointed in the right direction if they see this in CI. But pulling it up locally is likely going to be required to figure out the cause.

Instead of assertions in the before hook I could add a test that fails if the process exited with failure, but I think I'd have to define hooks/it blocks inside runScript 🤔